geode-protobuf/src/main/java/org/apache/geode/protocol/protobuf/utilities/ProtobufUtilities.java

```diff
@@ -133,21 +134,38 @@
 
   /**
    * This will return the object encoded in a protobuf EncodedValue
-   *
    * @param serializationService - object which knows how to encode objects for the protobuf
-   *        protocol {@link ProtobufSerializationService}
+   * protocol {@link ProtobufSerializationService}
    * @param encodedValue - The value to be decoded
    * @return the object encoded in the passed encodedValue
    * @throws UnsupportedEncodingTypeException - There isn't a SerializationType matching the
-   *         encodedValues type
+   * encodedValues type
    * @throws CodecNotRegisteredForTypeException - There isn't a protobuf codec for the
-   *         SerializationType matching the encodedValues type
+   * SerializationType matching the encodedValues type
    */
   public static Object decodeValue(SerializationService serializationService,
-      BasicTypes.EncodedValue encodedValue)
+                                   BasicTypes.EncodedValue encodedValue)
       throws UnsupportedEncodingTypeException, CodecNotRegisteredForTypeException {
     BasicTypes.EncodingType encoding = encodedValue.getEncodingType();
     byte[] bytes = encodedValue.getValue().toByteArray();
     return serializationService.decode(encoding, bytes);
   }
+
+  public static BasicTypes.Region createRegionMessageFromRegion(Region region) {
+    RegionAttributes regionAttributes = region.getAttributes();
+    BasicTypes.Region.Builder protoRegionBuilder = BasicTypes.Region.newBuilder();
+
+    protoRegionBuilder.setName(region.getName());
+    protoRegionBuilder.setSize(region.size());
+
+    protoRegionBuilder.setPersisted(regionAttributes.getDataPolicy().withPersistence());
+    protoRegionBuilder.setKeyConstraint(regionAttributes.getKeyConstraint() == null ? ""
```

Comment:
For when we don't have a key constaint, will protoRegionBuilder.setKeyConstraint("") do the same thing as simply not setting this field?